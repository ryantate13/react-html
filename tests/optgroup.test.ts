import {optgroup} from '../src';

describe('optgroup', () => {
    it('renders with children', () => {
        expect(optgroup({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(optgroup({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(optgroup()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(optgroup({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(optgroup({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(optgroup({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(optgroup({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(optgroup({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(optgroup({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(optgroup({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(optgroup({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(optgroup({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(optgroup({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(optgroup({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(optgroup({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(optgroup({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(optgroup({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(optgroup({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(optgroup({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(optgroup({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(optgroup({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(optgroup({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(optgroup({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(optgroup({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(optgroup({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(optgroup({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(optgroup({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(optgroup({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(optgroup({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(optgroup({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(optgroup({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(optgroup({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(optgroup({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(optgroup({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(optgroup({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(optgroup({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(optgroup({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(optgroup({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(optgroup({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(optgroup({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(optgroup({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(optgroup({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(optgroup({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(optgroup({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(optgroup({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(optgroup({translate: "no"})).toMatchSnapshot();
    });

    it('renders with disabled="true"', () => {
        expect(optgroup({disabled: true})).toMatchSnapshot();
    });

    it('renders with disabled="false"', () => {
        expect(optgroup({disabled: false})).toMatchSnapshot();
    });

    it('renders with label="TEST"', () => {
        expect(optgroup({label: "TEST"})).toMatchSnapshot();
    });
});