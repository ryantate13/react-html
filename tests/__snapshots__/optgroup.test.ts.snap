// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optgroup renders with accessKey="a,b,c,d" 1`] = `
<optgroup
  accessKey="a,b,c,d"
/>
`;

exports[`optgroup renders with autoCapitalize="characters" 1`] = `
<optgroup
  autoCapitalize="characters"
/>
`;

exports[`optgroup renders with autoCapitalize="none" 1`] = `
<optgroup
  autoCapitalize="none"
/>
`;

exports[`optgroup renders with autoCapitalize="off" 1`] = `
<optgroup
  autoCapitalize="off"
/>
`;

exports[`optgroup renders with autoCapitalize="on" 1`] = `
<optgroup
  autoCapitalize="on"
/>
`;

exports[`optgroup renders with autoCapitalize="sentences" 1`] = `
<optgroup
  autoCapitalize="sentences"
/>
`;

exports[`optgroup renders with autoCapitalize="words" 1`] = `
<optgroup
  autoCapitalize="words"
/>
`;

exports[`optgroup renders with children 1`] = `
<optgroup>
  TEST
</optgroup>
`;

exports[`optgroup renders with className="foo bar baz" 1`] = `
<optgroup
  className="foo bar baz"
/>
`;

exports[`optgroup renders with contentEditable="false" 1`] = `
<optgroup
  contentEditable="false"
/>
`;

exports[`optgroup renders with contentEditable="true" 1`] = `
<optgroup
  contentEditable="true"
/>
`;

exports[`optgroup renders with dir="auto" 1`] = `
<optgroup
  dir="auto"
/>
`;

exports[`optgroup renders with dir="ltr" 1`] = `
<optgroup
  dir="ltr"
/>
`;

exports[`optgroup renders with dir="rtl" 1`] = `
<optgroup
  dir="rtl"
/>
`;

exports[`optgroup renders with disabled="false" 1`] = `
<optgroup
  disabled={false}
/>
`;

exports[`optgroup renders with disabled="true" 1`] = `
<optgroup
  disabled={true}
/>
`;

exports[`optgroup renders with draggable="false" 1`] = `
<optgroup
  draggable="false"
/>
`;

exports[`optgroup renders with draggable="true" 1`] = `
<optgroup
  draggable="true"
/>
`;

exports[`optgroup renders with empty attributes 1`] = `<optgroup />`;

exports[`optgroup renders with hidden="false" 1`] = `
<optgroup
  hidden={false}
/>
`;

exports[`optgroup renders with hidden="true" 1`] = `
<optgroup
  hidden={true}
/>
`;

exports[`optgroup renders with id="TEST" 1`] = `
<optgroup
  id="TEST"
/>
`;

exports[`optgroup renders with inputMode="decimal" 1`] = `
<optgroup
  inputMode="decimal"
/>
`;

exports[`optgroup renders with inputMode="email" 1`] = `
<optgroup
  inputMode="email"
/>
`;

exports[`optgroup renders with inputMode="none" 1`] = `
<optgroup
  inputMode="none"
/>
`;

exports[`optgroup renders with inputMode="numeric" 1`] = `
<optgroup
  inputMode="numeric"
/>
`;

exports[`optgroup renders with inputMode="search" 1`] = `
<optgroup
  inputMode="search"
/>
`;

exports[`optgroup renders with inputMode="tel" 1`] = `
<optgroup
  inputMode="tel"
/>
`;

exports[`optgroup renders with inputMode="text" 1`] = `
<optgroup
  inputMode="text"
/>
`;

exports[`optgroup renders with inputMode="url" 1`] = `
<optgroup
  inputMode="url"
/>
`;

exports[`optgroup renders with is="foo" 1`] = `
<optgroup
  is="foo"
/>
`;

exports[`optgroup renders with itemID="https://google.com/" 1`] = `
<optgroup
  itemID="https://google.com/"
/>
`;

exports[`optgroup renders with itemProp="https://google.com/" 1`] = `
<optgroup
  itemProp="https://google.com/"
/>
`;

exports[`optgroup renders with itemRef="foo bar" 1`] = `
<optgroup
  itemRef="foo bar"
/>
`;

exports[`optgroup renders with itemScope="false" 1`] = `
<optgroup
  itemScope={false}
/>
`;

exports[`optgroup renders with itemScope="true" 1`] = `
<optgroup
  itemScope={true}
/>
`;

exports[`optgroup renders with itemType="https://google.com/" 1`] = `
<optgroup
  itemType="https://google.com/"
/>
`;

exports[`optgroup renders with label="TEST" 1`] = `
<optgroup
  label="TEST"
/>
`;

exports[`optgroup renders with lang="" 1`] = `
<optgroup
  lang=""
/>
`;

exports[`optgroup renders with lang="es-419" 1`] = `
<optgroup
  lang="es-419"
/>
`;

exports[`optgroup renders with slot="TEST" 1`] = `
<optgroup
  slot="TEST"
/>
`;

exports[`optgroup renders with spellCheck="false" 1`] = `
<optgroup
  spellCheck="false"
/>
`;

exports[`optgroup renders with spellCheck="true" 1`] = `
<optgroup
  spellCheck="true"
/>
`;

exports[`optgroup renders with style="[object Object]" 1`] = `
<optgroup
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`optgroup renders with tabIndex="1" 1`] = `
<optgroup
  tabIndex={1}
/>
`;

exports[`optgroup renders with title="TEST" 1`] = `
<optgroup
  title="TEST"
/>
`;

exports[`optgroup renders with translate="no" 1`] = `
<optgroup
  translate="no"
/>
`;

exports[`optgroup renders with translate="yes" 1`] = `
<optgroup
  translate="yes"
/>
`;

exports[`optgroup renders without attributes 1`] = `<optgroup />`;
